FirebaseVertexAI/Sources/VertexAI.swift

```diff
@@ -0,0 +1,112 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+import FirebaseAppCheckInterop
+import FirebaseCore
+
+// Exports the GoogleGenerativeAI module to users of the SDK.
+@_exported import GoogleGenerativeAI
+
+// Avoids exposing internal FirebaseCore APIs to Swift users.
+@_implementationOnly import FirebaseCoreExtension
+
+@available(iOS 15.0, macOS 11.0, macCatalyst 15.0, *)
+@objc(FIRVertexAI)
+open class VertexAI: NSObject {
+  // MARK: - Public APIs
+
+  /// Returns an instance of `GoogleGenerativeAI.GenerativeModel` that uses the Vertex AI API.
+  ///
+  /// This instance is configured with the default `FirebaseApp`.
+  public static func generativeModel(modelName: String, location: String) -> GoogleGenerativeAI
+    .GenerativeModel {
+    return generativeModel(app: FirebaseApp.app()!, modelName: modelName, location: location)
+  }
+
+  /// Returns an instance of `GoogleGenerativeAI.GenerativeModel` that uses the Vertex AI API.
+  public static func generativeModel(app: FirebaseApp, modelName: String,
+                                     location: String) -> GoogleGenerativeAI.GenerativeModel {
+    guard let provider = ComponentType<VertexAIProvider>.instance(for: VertexAIProvider.self,
+                                                                  in: app.container) else {
+      fatalError("No \(VertexAIProvider.self) instance found for Firebase app: \(app.name)")
+    }
+    let modelResourceName = modelResourceName(app: app, modelName: modelName, location: location)
+    let vertexAI = provider.vertexAI(location: location, modelResourceName: modelResourceName)
+
+    return vertexAI.model
+  }
+
+  // MARK: - Private
+
+  /// The `FirebaseApp` associated with this `VertexAI` instance.
+  private let app: FirebaseApp
+
+  private let appCheck: AppCheckInterop?
+
+  private let location: String
+
+  private let modelResouceName: String
+
+  lazy var model: GenerativeModel = {
+    let options = RequestOptions(
+      apiVersion: "v2beta",
+      endpoint: "staging-firebaseml.sandbox.googleapis.com",
+      hooks: [addAppCheckHeader]
+    )
+    return GenerativeModel(
+      name: modelResouceName,
+      apiKey: app.options.apiKey!,
+      requestOptions: options
+    )
+  }()
+
+  init(app: FirebaseApp, location: String, modelResourceName: String) {
+    self.app = app
+    appCheck = ComponentType<AppCheckInterop>.instance(for: AppCheckInterop.self, in: app.container)
+    self.location = location
+    modelResouceName = modelResourceName
+  }
+
+  private static func modelResourceName(app: FirebaseApp, modelName: String,
```

Comment:
Shouldn't this rather be `modelResourcePath`? In line 93, we return a path (or at least a path fragment).

Reply:
I chose this function name because it's a model resource name in resource-oriented design (see https://google.aip.dev/122).

Note: The field in the REST API is called [`model`](https://cloud.google.com/vertex-ai/docs/reference/rest/v1/projects.locations.publishers.models/streamGenerateContent#path-parameters) instead of `name`.